java/src/main/java/com/google/appengine/tools/pipeline/Job.java

```diff
@@ -379,12 +380,52 @@ private void setCurrentRunGuid(String guid) {
     return promisedValue;
   }
 
+  /**
+   * Invoke this method from within the {@code run} method of a <b>generator
+   * job</b> in order to declare that some value will be provided asynchronously
+   * by some external agent.
+   *
+   * @param <F> The type of the asynchronously provided value.
+   * @return A {@code PromisedValue} that represents an empty value slot that
+   *         will be filled at a later time when the external agent invokes
+   *         {@link PipelineService#submitPromisedValue(String, Object)}. This
+   *         may be passed in to further invocations of {@code futureCall()} in
+   *         order to specify a data dependency.
+   */
   public <F> PromisedValue<F> newPromise() {
     PromisedValueImpl<F> promisedValue =
         new PromisedValueImpl<>(getPipelineKey(), thisJobRecord.getKey(), currentRunGUID);
     updateSpec.getNonTransactionalGroup().includeSlot(promisedValue.getSlot());
     return promisedValue;
   }
+  
+  /**
+   * Invoke this method from within the {@code run} method of a <b>generator
+   * job</b> in order to get some value that has been previously declared and
+   * that will be provided asynchronously by some external agent. This can be
+   * used to share the same value with child {@link Job}s
+   * 
+   * 
+   * @param promiseHandle The unique identifier for the {@link PromisedValue}
+   *        obtained during the execution of some job via the method
```

Comment:
should we say "ancestor" job instead of "some" job? BTW, did you check what happens if called
by the same Job that invoked newPromise()? I would expect that not to work (as Slot is probably not saved until job runs complete...)